### PR TITLE
Add isPlanned to task

### PR DESCRIPTION
**Pull request by duanxianbo**
[(task:20610)](https://staging.teamyapp.com/teams/107/tasks/20610)

### CODE REVIEW

File: core/service/story.go [hunk 1]
```diff
@@ -302,7 +340,28 @@ func (s *Story) RemoveTaskFromStory(ct context.Context, storyID uint64, taskID u
...
 
-		return s.storyTaskRelationDao.DeleteStoryTaskRelation(ct, tx, storyID, taskID)
+		err = s.storyTaskRelationDao.DeleteStoryTaskRelation(ct, tx, storyID, taskID)
+		if err != nil {
+			return err
+		}
+
+		now := time.Now()
+		task.UpdatedAt = &now
+		task.IsPlanned = false
```
Comment (naturetime10) [inline core/service/story.go]: This assumes a task can only be in 1 story. I think we allow a task to be in multiple stories.

File: core/service/story.go [hunk 2]
```diff
@@ -334,6 +393,23 @@ func (s *Story) RemoveTasksFromStory(ct context.Context, storyID uint64, taskIDs
 			if err != nil {
 				return err
 			}
+
+			now := time.Now()
+			task.UpdatedAt = &now
+			task.IsPlanned = false
```
Comment (naturetime10) [inline core/service/story.go]: This assumes a task can only be in 1 story. I think we allow a task to be in multiple stories.

**Review by naturetime10 — CHANGES_REQUESTED** [review]

File: core/service/story.go (hunk 1 above)
Comment (duanxianbo) [reply core/service/story.go]: but that means, tasks will always be in the story task backlog?

File: core/service/story.go [hunk 3]
```diff
@@ -394,22 +402,29 @@ func (s *Story) RemoveTasksFromStory(ct context.Context, storyID uint64, taskIDs
...
+				now := time.Now()
+				task.UpdatedAt = &now
+				task.IsPlanned = false
+				updateTaskMutation := mutation.NewUpdateTask(
+					s.logger,
+					s.stateSyncer,
+					s.taskDao,
+					task,
+				)
+
```
Comment (naturetime10) [inline core/service/story.go]:
```suggestion
```

**Review by naturetime10 — APPROVED** [review]

**Pull request merged**